### ISSUE TITLE
Indent support

### ISSUE THREAD

**Issue by BerkeleyTrue** [othree/yajs.vim#20]
Wondering if indent support might be in the future for this project?

**Comment by othree** [comment]
I am willing to do it.
But I don't use fold function on Vim.
So I will need some one to help me to see is the logic good for people.

**Comment by shadowhand** [comment]
I'd love to have fold support on `{` if nothing else.

**Comment by HerringtonDarkholme** [comment]
@shadowhand  You can have a look at my fork. It supports TypeScript, a superset of JavaScript. So you can fork yats and modify it into a valid js syntax, with fold .
https://github.com/HerringtonDarkholme/yats.vim

**Comment by rschmukler** [comment]
As a temporary solution to anyone who wants it, I broke out the indenting from [pangloss/vim-javascript](https://github.com/pangloss/vim-javascript) (which I think is one of the better indenting detection libraries for javascript) and it is available [here](https://github.com/rschmukler/pangloss-vim-indent).

**Comment by rosshadden** [comment]
@rschmukler I like your idea, but cannot get your plugin to work.  Is there more setup required than just loading the plugin?

I'm asking you here instead of on your repo because it (probably) has to do with using it in conjunction with `yajs.vim`.  Thanks!

**Comment by othree** [comment]
I am using pangloss's indent file too :)

**Comment by rosshadden** [comment]
Yeah I was going to say, the indention in `yajs.vim` actually works perfectly as-is.  That's why I thought the OP was talking about supporting code folding.

**Comment by othree** [comment]
Since I don't use fold.
I am not sure which part people want to fold.
There is one vote for fold at `{`, `}`
But no other opinions~

**Comment by rosshadden** [comment]
I think we can close this, as the title is "Indent support", and as far as I am aware there are not actually any issues or holes with the indent support in this plugin.

I opened #49 with the fold support.

**Comment by olalonde** [comment]
@rosshadden did you do anything to get indentation support (I really mean indentation, not fold)? I'm seeing this with only the `othree/yajs` plugin enabled:

![sivcersgot](https://cloud.githubusercontent.com/assets/416585/19026689/949060cc-88dd-11e6-95de-d708f66fc5e9.gif)

Using `pangloss/vim-javascript` instead fixes the indentation issue.

**Comment by olalonde** [comment]
I just tried 

```javascript
Plugin 'othree/yajs.vim'
Plugin 'rschmukler/pangloss-vim-indent'
```

and indentation seems to work.

**Comment by rosshadden** [comment]
I use these relevant plugins:

```vim
Plug 'sheerun/vim-polyglot'
Plug 'othree/yajs.vim', { 'for': 'javascript' }
Plug 'othree/javascript-libraries-syntax.vim', { 'for': 'javascript' }
Plug 'othree/es.next.syntax.vim', { 'for': 'javascript' }
```

My behavior is the same as yours in your screenshot, but as far as I am concerned there is no problem.  You are missing a semicolon.  With the semicolon in place it works as expected, which is why I never noticed any issues.

**Comment by othree** [comment]
No semicolon is a coding style 😅 
ref: http://standardjs.com/

**Comment by olalonde** [comment]
Yes, I'm in the no semicolons camp :)
